test/java/src/com/ibm/streamsx/topology/test/spl/SPLOperatorsTest.java

```diff
@@ -119,6 +122,201 @@ public void testLiterals() throws Exception {
+        addParam("r", "rstring", r, params, submitParams);
+//        String u = "test \"Submission Parameters\"" + rand.nextInt();
+//        addParam("u", "ustring", u, params, submitParams);
+
+//      boolean bool = rand.nextBoolean();
+//      addParam("b", "boolean", bool, params, submitParams);
+      
+        
+        byte i8 = (byte) rand.nextInt();
+        short i16 = (short) rand.nextInt(); 
+        int i32 = rand.nextInt();
+        long i64 = rand.nextLong(); 
+        addParam("i8", "int8", i8, params, submitParams);
+        addParam("i16", "int16", i16, params, submitParams);
+        addParam("i32", "int32", i32, params, submitParams);
+        addParam("i64", "int64", i64, params, submitParams);
+        
+//        byte ui8 = (byte) rand.nextInt();
+//        short ui16 = (short) rand.nextInt(); 
+//        int ui32 = rand.nextInt();
+//        long ui64 = rand.nextLong(); 
+//        addParam("ui8", "uint8", ui8, params, submitParams);
+//        addParam("ui16", "uint16", ui16, params, submitParams);
+//        addParam("ui32", "uint32", ui32, params, submitParams);
+//        addParam("ui64", "uint64", ui64, params, submitParams);
+        
+        float f32 = rand.nextFloat();
+        double f64 = rand.nextDouble();
+        addParam("f32", "float32", f32, params, submitParams);
+        addParam("f64", "float64", f64, params, submitParams);
+        
+        getConfig().put(ContextProperties.SUBMISSION_PARAMS, submitParams);
+   
+        SPL.addToolkit(topology, new File(getTestRoot(), "spl/testtk"));
+        SPLStream paramTuple = SPL.invokeSource(topology, "testgen::TypeLiteralTester", params, schema);
+
+        Tester tester = topology.getTester();
+        
+        Condition<Long> expectedCount = tester.tupleCount(paramTuple, 1);
+        MostRecent<Tuple> mr = tester.splHandler(paramTuple, new MostRecent<Tuple>());
+
+        complete(tester, expectedCount, 10, TimeUnit.SECONDS);
+
+        assertTrue(expectedCount.toString(), expectedCount.valid());
+        Tuple tuple = mr.getMostRecentTuple();
+        
+//        assertEquals(bool, tuple.getBoolean("bool"));
+        assertEquals(r, tuple.getString("r"));
+//        assertEquals(u, tuple.getString("u"));
+        
+        assertEquals(i8, tuple.getByte("i8"));
+        assertEquals(i16, tuple.getShort("i16"));
+        assertEquals(i32, tuple.getInt("i32"));
+        assertEquals(i64, tuple.getLong("i64"));
+        
+//        assertEquals(ui8, tuple.getByte("ui8"));
+//        assertEquals(ui16, tuple.getShort("ui16"));
+//        assertEquals(ui32, tuple.getInt("ui32"));
+//        assertEquals(ui64, tuple.getLong("ui64"));
+
+        assertEquals("f32="+f32, f32, tuple.getFloat("f32"), 0.001);
+        assertEquals("f64="+f64, f64, tuple.getDouble("f64"), 0.001);
+    }
+
+    static void addParamDefault(String name, String splType, String defaultVal, Map<String,Object> params)
+    {
+        String spName = name + ".param";
```

Comment:
Why is the name of the parameter different in the map passed in at submission time?